sysrepo does not build on OpenWrt with an older uClibc toolchain

### Description
Sysrepo has 3 bugs when build with older uClibc toolchain:

1)
```
/opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_utils.h:182:48: error: unknown type name 'va_list'
 int sr_vasprintf(char **strp, const char *fmt, va_list ap);
                                                ^
In file included from /opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_common.h:49:0,
                 from /opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_common.c:28:
/opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_mem_mgmt.h:152:86: error: unknown type name 'va_list'
 int sr_mem_edit_string_va(sr_mem_ctx_t *sr_mem, char **string_p, const char *format, va_list args);
                                                                                      ^
make[5]: *** [src/CMakeFiles/COMMON.dir/common/sr_common.c.o] Error 1
```

2)
```
/opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_utils.h:182:48: error: unknown type name 'va_list'
 int sr_vasprintf(char **strp, const char *fmt, va_list ap);
                                                ^
In file included from /opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_common.h:49:0,
                 from /opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_common.c:28:
/opt/inteno/iopsys/build_dir/target-arm_xscale_uClibc-0.9.33.2_eabi/sysrepo-e4a5705a4c5293ba121048c71899b98243d99edf/src/common/sr_mem_mgmt.h:152:86: error: unknown type name 'va_list'
 int sr_mem_edit_string_va(sr_mem_ctx_t *sr_mem, char **string_p, const char *format, va_list args);
                                                                                      ^
make[5]: *** [src/CMakeFiles/COMMON.dir/common/sr_common.c.o] Error 1
```

3)
```
Linking C executable sysrepo-plugind
libsysrepo.so.0.6.0: undefined reference to `mkstemps'
collect2: error: ld returned 1 exit status
```

--- a/src/clientlib/cl_subscription_manager.c
+++ b/src/clientlib/cl_subscription_manager.c
@@ -1454,22 +1454,17 @@ cl_sm_get_server_socket_filename(cl_sm_ctx_t *sm_ctx, const char *module_name, c
     /* append temporary file name part */
     strncat(path, ".XXXXXX.sock", PATH_MAX - strlen(path) - 1);
     fd = mkstemps(path, 5);
-    if (-1 == fd) {
-        close(fd);
-        unlink(path);
-    }
 #else
     /* append temporary file name part */
     strncat(path, ".XXXXXX", PATH_MAX - strlen(path) - 1);
-    if (mktemp(path)) {
-        strncat(path, ".sock", PATH_MAX - strlen(path) - 1);
-        fd = open(path, O_RDWR | O_CREAT | O_EXCL, 0600);
-        if (-1 == fd) {
-            close(fd);
-            unlink(path);
-        }
-    }
+    /* cannot fail */
+    mktemp(path);
+    strncat(path, ".sock", PATH_MAX - strlen(path) - 1);
+    fd = open(path, O_RDWR | O_CREAT | O_EXCL, 0600);
 #endif
+    CHECK_NOT_MINUS1_LOG_RETURN(fd, -1, "Failed to open unique temporary file: %s", strerror(errno));  
+    close(fd);
+    unlink(path);
 
     *socket_path = strdup(path);
     CHECK_NULL_NOMEM_RETURN(*socket_path);

--- a/src/clientlib/cl_subscription_manager.c
+++ b/src/clientlib/cl_subscription_manager.c
@@ -31,6 +31,9 @@
 #include <pthread.h>
 #include <arpa/inet.h>
 #include <ev.h>
+#ifndef HAVE_MKSTEMPS
+#include <fcntl.h>
+#endif
 
 #include "cl_subscription_manager.h"
 #include "sr_common.h"
@@ -1447,13 +1450,26 @@ cl_sm_get_server_socket_filename(cl_sm_ctx_t *sm_ctx, const char *module_name, c
     snprintf(pid_str, 20, "%d", getpid());
     strncat(path, pid_str, PATH_MAX - strlen(path) - 1);
 
+#ifdef HAVE_MKSTEMPS
     /* append temporary file name part */
     strncat(path, ".XXXXXX.sock", PATH_MAX - strlen(path) - 1);
     fd = mkstemps(path, 5);
-    if (-1 != fd) {
+    if (-1 == fd) {
         close(fd);
         unlink(path);
     }
+#else
+    /* append temporary file name part */
+    strncat(path, ".XXXXXX", PATH_MAX - strlen(path) - 1);
+    if (mktemp(path)) {
+        strncat(path, ".sock", PATH_MAX - strlen(path) - 1);
+        fd = open(path, O_RDWR | O_CREAT | O_EXCL, 0600);
+        if (-1 == fd) {
+            close(fd);
+            unlink(path);
+        }
+    }
+#endif
 
     *socket_path = strdup(path);
     CHECK_NULL_NOMEM_RETURN(*socket_path);

--- a/src/common/sr_utils.c
+++ b/src/common/sr_utils.c
@@ -455,14 +455,14 @@ sr_fd_set_nonblock(int fd)
 
 #if defined(SO_PEERCRED)
 
-#if !defined(__USE_GNU)
+#if !defined(SCM_CREDENTIALS)
 /* struct ucred is ifdefined behind __USE_GNU, but __USE_GNU is not defined */
 struct ucred {
     pid_t pid;    /* process ID of the sending process */
     uid_t uid;    /* user ID of the sending process */
     gid_t gid;    /* group ID of the sending process */
 };
-#endif /* !defined(__USE_GNU) */
+#endif /* !defined(SCM_CREDENTIALS) */
 
 int
 sr_get_peer_eid(int fd, uid_t *uid, gid_t *gid)

--- a/src/common/sr_utils.h
+++ b/src/common/sr_utils.h
@@ -23,6 +23,7 @@
 #ifndef SR_UTILS_H_
 #define SR_UTILS_H_
 
+#include <stdarg.h>
 #include <stdio.h>
 #include <stdbool.h>
 #define __USE_XOPEN

